keps/sig-node/4191-split-image-filesystem/README.md

```diff
@@ -0,0 +1,891 @@
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP is about enhancing Kubelet to be aware if a container runtime splits the image filesystem.  
+Aware in this case means that garbage collecting images, containers and reporting disk usage is all functional.
+
+## Motivation
+
+Kubelet has two distinct filesystems: Node and Image. In typical deployments, users deploy Kubernetes where both the Node and Image filesystems are on the same disk. There are some requests to separate the storage into separate disks.  The most common request is to separate the writable layer from the read-only layer. Kubelet and Container data would be stored on the same disk while images would have their own disk.  This could be beneficial because images occupy a lot of disk space while the writeable layer is typically smaller.
+
+Container IO can impact Kubelet and adding the ability for more disks could increase performance of Kubelet.
+
+However, it is not possible to separate the image layers and container writable layers on different disks.
+
+In the current implementation of separate disks, containers and images must be stored on the same disk.  So garbage collection, in case of node pressure (really image disk pressure) would GC images/containers on the image filesystem.
+
+If one separates writable layer (containers) from readable layer (images), then garbage collection and statistics must account for this separation.  Today this could potentially break Kubelet if the container runtime configures storage in this way.
+
+One downside of the separate disk is that pod data can be written in multiple locations.  The writeable layer of a container would go on the image filesystem and volume storage would go to the root fs.  There is another request to separate the root and the image filesystem to be writeable and read-only respective.  This means that pod data can be written on one disk while the other disk can be read-only.  Separting the writeable layer and the read-only layer will achieve this.
+
+### Goals
+
+- Kubelet should still work if images/containers are separated into different disks.
+  - Support writable layer being on same disk as Kubelet
+  - Images can be on the separate filesystem.
+
+### Possible Extensions in Post Alpha
+
+- Kubelet, Images and Containers on all separate disks.
```

Comment:
from our convo at WG meeting, isn't this in scope for alpha now?

Reply:
I thought we agree to go forward with alpha for writeable on root and read-only on imagefs.  We can support this but we won't say that we are testing or expect support for it.

We won't be supporting multiple disks but other cases are possible but we are not explicity testing them?  Hence why I said possible extensions in case there is interest in these use cases and time for people to explore them.

Reply:
We followed up offline.

Added a note to make this clear.